torch/utils/mkldnn.py

```diff
@@ -104,6 +104,56 @@ def __setstate__(self, state):
         self.bias = state[1].to_mkldnn()
         self.training = state[2]
 
+class MkldnnConv3d(torch.jit.ScriptModule):
+    __constants__ = ['stride', 'padding', 'dilation', 'groups']
+
+    def __init__(self, dense_module):
+        super(MkldnnConv3d, self).__init__()
+
+        self.stride = dense_module.stride
+        self.padding = dense_module.padding
+        self.dilation = dense_module.dilation
+        self.groups = dense_module.groups
+
+        self.register_buffer('weight', torch._C._nn.mkldnn_reorder_conv3d_weight(
+            dense_module.weight.to_mkldnn(),
+            self.padding,
+            self.stride,
+            self.dilation,
+            self.groups))
+        if dense_module.bias is not None:
+            self.register_buffer('bias', dense_module.bias.to_mkldnn())
+        else:
+            # TODO: Remove this once ScriptModule supports registering None buffer
+            self.register_buffer(
+                'bias',
+                torch.zeros([dense_module.weight.size(0)], dtype=torch.float).to_mkldnn())
+
+    @torch.jit.script_method
+    def __getstate__(self):
+        return (self.weight.to_dense(), self.bias.to_dense(), self.training)
+
+    @torch.jit.script_method
+    def __setstate__(self, state):
+        self.weight = torch._C._nn.mkldnn_reorder_conv3d_weight(
+            state[0].to_mkldnn(),
+            self.padding,
+            self.stride,
+            self.dilation,
+            self.groups)
+        self.bias = state[1].to_mkldnn()
+        self.training = state[2]
+
+    @torch.jit.script_method
+    def forward(self, x):
+        return torch.conv3d(
```

Comment:
For the first question, DNNL also support dilation for convNd, I can enable it, so there only has one case not supported by DNNL: x is not a float tensor, but for this case,  I think weight is also a float tensor, it will report an error to user when reorder the weight, because it need to call **.to_mkldnn()**first which will check the tensor's type.

For the second question, yes, the name is confused, I will change it.

Reply:
Ok, please enable dilated convolution then. I agree that for correct user inputs this situation should not happen, but even in case of incorrect user inputs (user had float weight when creating a module, but is sending double tensor, or cuda tensor, to forward) the error message should be clear and helpful, and I don't know what will happen in this case.

Reply:
DNNL dilation conv is enabled now. thanks!